Src/AutoFixture.NUnit3/AutoDataAttribute.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Builders;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture.NUnit3
+{
+    /// <summary>
+    /// This attribute uses AutoFixture to generate values for unit test parameters. 
+    /// This implementation is based on TestCaseAttribute of NUnit3
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method)]
+    public class AutoDataAttribute : Attribute, ITestBuilder
+    {
+        private readonly IFixture _fixture;
+
+        /// <summary>
+        /// Construct a <see cref="AutoDataAttribute"/>
+        /// </summary>
+        public AutoDataAttribute()
+            : this(new Fixture())
+        {
+        }
+
+        /// <summary>
+        /// Construct a <see cref="AutoDataAttribute"/> with an <see cref="IFixture"/> 
+        /// </summary>
+        /// <param name="fixture"></param>
+        protected AutoDataAttribute(IFixture fixture)
```

Comment:
Ah, good point. Should we do something similar for `[AutoData]` in _AutoFixture.Xunit2_, then?